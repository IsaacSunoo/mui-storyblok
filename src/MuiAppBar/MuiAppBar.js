import React, { createElement, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Storyblok from '../utils/Storyblok';

const MuiTypography = lazy(() => import('../MuiTypography/MuiTypography'));
const MuiButtonRedirect = lazy(() => import('../MuiButtonRedirect/MuiButtonRedirect'));
// const MuiMenu = lazy(() => import('sharedComponents/StoryBlok/MuiMenu/MuiMenu'));

const MuiAppBar = ({
  rootClass,
  color,
  position,
  content,
  variant,
  disableGutters,
}) => {
  const components = {
    MuiTypography,
    MuiButtonRedirect,
    // MuiMenu,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AppBar
      className={styles.root}
      color={color}
      position={position}
    >
      <Toolbar
        variant={variant}
        disableGutters={disableGutters}
      >
        <Suspense fallback={<div />}>
          {content.map((item, index) => createElement(
            components[item.component],
            Object.assign(item, { key: index }),
          ))}
        </Suspense>
      </Toolbar>
    </AppBar>
  );
};

export default MuiAppBar;

MuiAppBar.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** 'default' | 'inherit' | 'primary' | 'secondary' | 'transparent' */
  color: PropTypes.string,
  /** 'absolute'| 'fixed'| 'relative'| 'static' | 'sticky' */
  position: PropTypes.string,

  /** mui prop: 'regular'| 'dense' */
  variant: PropTypes.string,
  /** mui prop: true | false */
  disableGutters: PropTypes.bool,


  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiAppBar.defaultProps = {
  color: 'primary',
  position: 'fixed',
  variant: 'regular',
  disableGutters: false,
  rootClass: [],
};
