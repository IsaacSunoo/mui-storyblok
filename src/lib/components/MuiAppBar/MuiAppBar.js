import React, { createElement, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Storyblok from '../../utils/Storyblok';

const MuiTypography = lazy(() => import('../MuiTypography/MuiTypography'));
const MuiIconButtonRedirect = lazy(() => import('../MuiIconButtonRedirect/MuiIconButtonRedirect'));
const MuiIconButtonHref = lazy(() => import('../MuiIconButtonHref/MuiIconButtonHref'));
const MuiButtonRedirect = lazy(() => import('../MuiButtonRedirect/MuiButtonRedirect'));
const MuiButtonHref = lazy(() => import('../MuiButtonHref/MuiButtonHref'));
const MuiMenu = lazy(() => import('../MuiMenu/MuiMenu'));

const MuiAppBar = ({
  rootClass,
  color,
  position,
  content,
  variant,
  disableGutters,
}) => {
  const components = {
    MuiTypography,
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiMenu,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AppBar
      className={styles.root}
      color={color}
      position={position}
    >
      <Toolbar
        variant={variant}
        disableGutters={disableGutters}
      >
        <Suspense fallback={<div />}>
          {content.map((item, index) => createElement(
            components[item.component],
            Object.assign(item, { key: index }),
          ))}
        </Suspense>
      </Toolbar>
    </AppBar>
  );
};

export default MuiAppBar;

MuiAppBar.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component.
  */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: 'default' | 'inherit' | 'primary' | 'secondary' | 'transparent' 
   * AppBar: The color of the component. It supports those theme colors that make sense for this component.
   * */
  color: PropTypes.string,
  /**
   * 'absolute'| 'fixed'| 'relative'| 'static' | 'sticky'
   * AppBar: The positioning type. The behavior of the different options is described in the MDN web docs.
   * Note: sticky is not universally supported and will fall back to static when unavailable.
   * */
  position: PropTypes.string,
  /**
   * mui prop: 'regular'| 'dense'
   * ToolBar: The variant to use.
   *  */
  variant: PropTypes.string,
  /**
   * mui prop: true | false
   * Toolbar: If true, disables gutter padding.
   * */
  disableGutters: PropTypes.bool,

  /**
   * Content passed to MuiGrid to render
   * components: MuiTypography,
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiMenu,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiAppBar.defaultProps = {
  color: 'primary',
  position: 'fixed',
  variant: 'regular',
  disableGutters: false,
  rootClass: [],
};
