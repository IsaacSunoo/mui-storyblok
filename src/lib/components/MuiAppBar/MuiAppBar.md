MuiAppBar example:

```js noeditor
import MuiAppBar from './MuiAppBar';
```

```js
const content = [{
  component: 'MuiButtonRedirect',
  redirectRoute: '',
  button: [{
    component: 'MuiButton',
    buttonText: 'here',
  }],
}];

  <MuiAppBar
    position="relative"
    content={content}
  />
```