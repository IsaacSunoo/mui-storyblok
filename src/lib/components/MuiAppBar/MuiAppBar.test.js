import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiAppBar from './MuiAppBar';

function setup() {
  const props = {
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        text: 'this is text',
      }],
    }],
  };
  const comp = shallow(<MuiAppBar {...props} />);
  return { comp, props };
}

describe('<MuiAppBar />', () => {
  it('renders MuiAppBar', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiAppBar {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
