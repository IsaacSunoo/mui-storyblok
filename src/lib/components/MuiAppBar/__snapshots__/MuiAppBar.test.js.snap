// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiAppBar /> snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-root-3 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <div />
  </div>
</header>
`;
